packages/truffle-core/lib/commands/develop.js

```diff
@@ -14,104 +15,79 @@ var command = {
 
-    var available_commands = Object.keys(commands).filter(function(name) {
-      return excluded.indexOf(name) === -1;
-    });
+    const available_commands = Object.keys(commands).filter(
+      name => !excluded.includes(name)
+    );
 
-    var console_commands = {};
-    available_commands.forEach(function(name) {
+    const console_commands = {};
+    available_commands.forEach(name => {
       console_commands[name] = commands[name];
     });
 
-    Environment.develop(config, ganacheOptions, function(err) {
-      if (err) return done(err);
-
-      var c = new Console(
-        console_commands,
-        config.with({
-          noAliases: true
-        })
-      );
+    const environmentDevelop = util.promisify(Environment.develop);
+    environmentDevelop(config, ganacheOptions).catch(err => done(err));
 
-      c.start(done);
-      c.on("exit", function() {
-        process.exit();
-      });
+    const c = new Console(console_commands, config.with({ noAliases: true }));
+    c.start(done);
+    c.on("exit", async () => {
+      process.exit();
     });
   },
-  run: function(options, done) {
-    var Config = require("truffle-config");
-    var Develop = require("../develop");
-
-    var config = Config.detect(options);
-    var customConfig = config.networks.develop;
-    let numAddresses = 10;
-    let defaultEtherBalance = 100;
-    let bTime = 0;
-
-    if (customConfig) {
-      numAddresses = customConfig.accounts ? customConfig.accounts : 10;
-      defaultEtherBalance = customConfig.defaultEtherBalance
-        ? customConfig.defaultEtherBalance
-        : 100;
-      bTime = customConfig.blockTime ? customConfig.blockTime : 0;
-    }
+  run: async (options, done) => {
+    const Config = require("truffle-config");
+    const Develop = require("../develop");
+
+    const config = Config.detect(options);
+    const customConfig = config.networks.develop || {};
 
     const { mnemonic, accounts, privateKeys } = mnemonicInfo.getAccountsInfo(
-      numAddresses
+      customConfig.accounts || 10
     );
 
-    var onMissing = function(name) {
-      return "**";
-    };
+    const onMissing = name => "**";
 
-    var warning =
+    const warning =
       ":warning:  Important :warning:  : " +
       "This mnemonic was created for you by Truffle. It is not secure.\n" +
       "Ensure you do not use it on production blockchains, or else you risk losing funds.";
 
-    var ipcOptions = {
+    const ipcOptions = {
       log: options.log
     };
 
-    var ganacheOptions = {
-      host: "127.0.0.1",
-      port: 9545,
-      network_id: 4447,
-      total_accounts: numAddresses,
-      default_balance_ether: defaultEtherBalance,
-      blockTime: bTime,
-      mnemonic: mnemonic,
-      gasLimit: config.gas,
+    const ganacheOptions = {
+      host: customConfig.host || "127.0.0.1",
+      port: customConfig.port || 9545,
+      network_id: customConfig.network_id || 5777,
```

Comment:
(I don't think we should change this from 4447 just because of MetaMask. We can work with them to get a better solution)

Reply:
Pulling out the `network_id` specific commit